README.md

```diff
@@ -1 +1,145 @@
+
+Good luck and happy coding! 🚀
+
+---
+
+## Exercise
+
+### TASK #1 - JavaScript
+
+Implement `prepareData` higher order function, which takes an object of filter params `{year, customerName}`, and returns a function that processes a list of missions, by only showing the ones that were launched in `year` and carried a payload belonging to `customerName`.
+
+**Observations:**
+
+- Missions should appear in inverse chronological order (sort), with the exception of those that carried more payloads should appear first.
+- Payloads are carried in the second stage of a rocket and they can belong to multiple customers.
+- It doesn't matter to which `customerName` 'program' each payload belongs to as long as `customerName` is the customer.
+
+**Example:**
+
+Considering we have the list of missions from this [data fixture][data-fixtures], and the following filter params:
+
+```js
+{
+  year: 2018,
+  customerName: "NASA"
+}
+```
+
+The expected result should be:
+
+```js
+[
+  {
+    flight_number: 62,
+    mission_name: "Iridium NEXT Mission 6",
+    payloads_count: 2,
+  },
+  {
+    flight_number: 72,
+    mission_name: "CRS-16",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 64,
+    mission_name: "CRS-15",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 60,
+    mission_name: "TESS",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 59,
+    mission_name: "CRS-14",
+    payloads_count: 1,
+  },
+];
+```
+
+---
+
+### TASK #2 - React & Hooks
+
+Implement `<RocketsList>` component with the following specifications:
+
+- The component receives `filerParams` object (with the shape described in [task #1][task-1]) as a prop.
+- The component obtains a list of 'missions' from a [custom hook][custom-hook], which fetches **the whole list of missions** from the [SpaceX API][spacex-api] and processes them with `prepareData` function (from [task #1][task-1]) and `filterParams` prop.
+  - The mission data should only be fetched once.
+  - As a part of this challenge, you are not allowed to use any of the filter parameters provided by the [SpaceX API docs][spacex-api-docs].
+- For each 'mission' obtained from the custom hook, the component renders a string using [template literals][template-literals] with the following format: "#`flight_number` `mission_name` (`payloads_count`)"
+- While 'missions' are being fetched from the API, the component renders `"Loading..."` on the screen.
+- If no 'missions' are obtained from the custom hook, the component renders `"No data"` on the screen.
+
+**Example:**
+
+Considering we pass the following `filterParams` as a prop to the `RocketList` component:
+
+```js
+{
+  year: 2018,
+  customerName: "NASA"
+}
+```
+
+The expected render should be similar to the following:
+
+<img align="center" src="img/expected_output.png" width="600">
+
+We will be testing your solution with a small set of E2E tests, please make sure your solution conforms to the specifications above.
```

Comment:
let's make sure that e2e covers couple of cases we've mentioned - no data, loading, different data cases, including the one on the screenshot

Reply:
not sure how to catch where they export `RocketList`, maybe we can ask them to export it by a specific path for easier checking?

Reply:
> not sure how to catch where they export RocketList, maybe we can ask them to export it by a specific path for easier checking?

I was also thinking about this one. Maybe we should just specify that their solution app should render the example above, with fixed `filterParams`? We would still be able to check loading state and final result.

Reply:
Another idea, which might complicate the home-task a bit – `filterParam` can be specified via the URL. This way we'd be able to do extensive testing of the solution with E2E.

Reply:
Sounds like fun, but would make it a bit harder indeed. I wonder if we can A/B test two version to see if it worth it :D

But in general, I would prefer catching their export or URL approach, since rendering with fixed `filterParams` would for sure lead to false positives where rendering is correct, but internals aren't (hardcoded or smth). Ideally we have e2e to catch such cases before reviewer even have to spend their time reviewing code.

Reply:
To add:
* Use `<ul>` and `<li>`
* `Loading...` must be exact copy.
* Should expose `npm run start`
* Should run on `localhost:3000`.